anndata/tests/test_x.py

```diff
@@ -30,11 +32,16 @@ def diskfmt(request):
 @pytest.mark.parametrize("shape", SINGULAR_SHAPES)
 @pytest.mark.parametrize("orig_array_type", UNLABELLED_ARRAY_TYPES)
 @pytest.mark.parametrize("new_array_type", UNLABELLED_ARRAY_TYPES)
-def test_setter_singular_dim(shape, orig_array_type, new_array_type):
+def test_setter_singular_dim(shape, orig_array_type, new_array_type, request):
     # https://github.com/scverse/anndata/issues/500
     adata = gen_adata(shape, X_type=orig_array_type)
     adata.X = new_array_type(np.ones(shape))
     np.testing.assert_equal(asarray(adata.X), 1)
+    id = request.node.callspec.id.split("-")[0]
+    if id == "ndarray":
+        assert isinstance(adata.X, np.ndarray)
+    else:
+        assert isinstance(adata.X, new_array_type)
```

Comment:
`X` is cast immediately after that to `new_array_type`, which is what the issue that this links to says we should be loooking for, no?

Reply:
I meant for `id`

Reply:
ok, we can do that.

Reply:
Nevermind, it's better to use the `id` because some of these `new_array_type` are functions, which have `__name__` and others are not since they come from `partial`.